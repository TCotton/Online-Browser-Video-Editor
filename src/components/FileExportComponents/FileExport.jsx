import React from 'react'
import {
  Dialog,
  DialogBackdrop,
  DialogDisclosure,
  useDialogState
} from 'reakit/Dialog'
import { css } from '@emotion/css'
import { useDispatch } from 'react-redux'
import { FileConvertExport } from './FileConvertExport'
import { stopFn } from '../slices/playerSlice'

const backdropStyles = css`
  perspective: 800px;
  transition: opacity 250ms ease-in-out;
  opacity: 1;
  background-color: rgba(0, 0, 0, 0.5);
  position: fixed;
  inset: 0;
  z-index: 999;

  &[data-enter] {
    opacity: 1;
  }
`

const dialogueStyles = css`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`

export const FileExport = () => {
  const dialog = useDialogState({ baseId: 'FileExport' })
  const ref = React.useRef()
  const dispatch = useDispatch()
  React.useEffect(() => {
    if (dialog.visible) {
      dispatch(stopFn(true))
    }
  }, [dialog.visible])
  return (
    <>
      <DialogDisclosure {...dialog}>
        <img
          src={'./export.svg'}
          className='export'
          alt='export'
          data-testid='file-export'
        />
      </DialogDisclosure>
      <DialogBackdrop {...dialog} className={backdropStyles}>
        <Dialog {...dialog} aria-label='Welcome' className={dialogueStyles}>
          <div ref={ref}>
            <FileConvertExport />
          </div>
        </Dialog>
      </DialogBackdrop>
    </>
  )
}
