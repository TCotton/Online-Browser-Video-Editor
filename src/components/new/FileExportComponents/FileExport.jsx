import React from 'react';

export const FileExport = () => {
    return (
        <img src={"./export.svg"} className="export" alt="export" data-testid="file-export"/>
    )
}