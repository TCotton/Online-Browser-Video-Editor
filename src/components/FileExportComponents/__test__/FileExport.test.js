import React from 'react'
import { render } from '@testing-library/react'
import { FileExport } from '../FileExport'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn()
}))

jest.mock('dexie', () => {
  return class Dexie {
    open () {
      return Promise.resolve()
    }
    version () {
      return {
        stores: jest.fn()
      }
    }
  }
})

jest.mock('dexie-react-hooks', () => ({
  ...jest.requireActual('dexie-react-hooks'),
  useLiveQuery: jest.fn()
}))

describe('FileExport', () => {
  it('should render successfully', () => {
    const { container } = render(<FileExport />)
    expect(
      container.querySelector('[data-testid="file-export"]')
    ).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<FileExport />)
    expect(baseElement).toMatchSnapshot()
  })
})
