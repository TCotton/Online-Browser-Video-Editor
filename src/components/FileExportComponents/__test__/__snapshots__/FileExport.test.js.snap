// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileExport should match snapshot 1`] = `
<body>
  <div>
    <button
      aria-controls="FileExport"
      aria-expanded="false"
      aria-haspopup="dialog"
      type="button"
    >
      <img
        alt="export"
        class="export"
        data-testid="file-export"
        src="./export.svg"
      />
    </button>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      class="css-jhczzu"
      data-dialog-ref="FileExport"
      hidden=""
      style="display: none;"
    >
      <div
        aria-label="Welcome"
        aria-modal="true"
        class="css-jkbxbt"
        data-dialog="true"
        hidden=""
        id="FileExport"
        role="dialog"
        style="display: none;"
        tabindex="-1"
      >
        <div>
          <div
            class="export"
            data-testid="export"
          >
            <header>
              <p
                class="title"
              >
                Click below to export
              </p>
              <p
                class="status"
              />
            </header>
            <ul
              class="social"
            >
              <li>
                <button
                  data-testid="twitter"
                >
                  <img
                    alt="click to export for twitter"
                    height="400"
                    src="./social/twitter.svg"
                    width="400"
                  />
                </button>
              </li>
              <li>
                <button
                  data-testid="facebook"
                >
                  <img
                    alt="click to export for Facebook"
                    height="250"
                    src="./social/facebook.png"
                    width="250"
                  />
                </button>
              </li>
              <li>
                <button
                  data-testid="linkedin"
                >
                  <img
                    alt="click to export for Linkedin"
                    height="540"
                    src="./social/linkedin.png"
                    width="650"
                  />
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
